migrator/src/test/scala/akka/persistence/jdbc/migrator/BaseJournalMigratorSpec.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2014 - 2019 Dennis Vriend <https://github.com/dnvriend>
+ * Copyright (C) 2019 - 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.jdbc.migrator
+
+import akka.persistence.jdbc.SingleActorSystemPerTestSpec
+import akka.persistence.jdbc.journal.dao.legacy
+import akka.persistence.PersistentRepr
+import akka.serialization.Serialization
+import akka.Done
+import slick.jdbc.JdbcBackend
+
+import java.util.UUID
+import scala.concurrent.{ ExecutionContext, Future }
+
+abstract class BaseJournalMigratorSpec(config: String) extends SingleActorSystemPerTestSpec(config) {
```

Comment:
I think it will be more convenient (and closer to user experience) if the tests start a `EventSourcedBehavior` using the legacy tables and fill it by sending commands instead of going low-level and manipulate the queries directly. 

When doing so, we can also configure the behavior to save the snapshots every N events (eg: N=5) and to keep 3 snapshots for example. 

That's why I was saying that we may need to start two actor systems. 

I suggest the following test plan: 

1. start actor system with the legacy config and populate the journal for a few entities, with many events and snapshots. Stop actor system
2. run migrator for events and snapshots
3. start another actor system using default config. Reload each entity and verify that their state is as expected. 

The Account entity is a good one because we can easily test the balance.

Another tests could be: 

1. start actor system with the legacy config and populate the journal for a few entities, with many events and **save snapshot on each event**. Stop actor system
2. run migrator only for snapshots
3. start another actor system using default config. Reload each entity and verify that they state is as expected. 

In that second test, since we save snapshots on each event we don't need to migrate the events and when verifying the state, we prove that we indeed migrated all snapshots.

Finally, we need to test that the tags are correctly migrated. 
For that test I think it suffice to create one entity that persist events with tags, migrate the events and consume the tags using `eventsByTags` from the new table. 

In that test, we can simply verify that we consume the same amount of events as we generated through the entity. 

WDYT?